storage/aptosdb/src/state_store/state_merkle_batch_committer.rs

```diff
@@ -62,15 +58,10 @@ impl StateMerkleBatchCommitter {
                         .state_merkle_db
                         .write_schemas(batch)
                         .expect("State merkle batch commit failed.");
-                    if self.state_db.state_merkle_db.cache_enabled() {
-                        self.state_db
-                            .state_merkle_db
-                            .version_cache()
-                            .maybe_evict_version(self.state_db.state_merkle_db.lru_cache());
-                    }
-                    // TODO(grao): Consider remove the following sender once we verified the
-                    // version cache correctly cached all nodes we need.
-                    snapshot_ready_sender.send(()).unwrap();
+                    self.state_db
+                        .state_merkle_db
+                        .version_cache()
+                        .maybe_evict_version(self.state_db.state_merkle_db.lru_cache());
```

Comment:
how do we make sure cached versions > versions not yet committed?

Reply:
Put an assert! in state_snapshot_committer.rs to ensure we cached more batches than pending commit batches.